plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt

```diff
@@ -622,27 +622,23 @@ class CodeWhispererService : Disposable {
         // the upper bound for supplemental context duration is 50ms
         // 2. supplemental context
         val startFetchingTimestamp = System.currentTimeMillis()
-        val isTstFile = FileContextProvider.getInstance(project).isTestFile(psiFile)
         val supplementalContext = runBlocking {
-            try {
-                withTimeout(SUPPLEMENTAL_CONTEXT_TIMEOUT) {
-                    FileContextProvider.getInstance(project).extractSupplementalFileContext(psiFile, fileContext)
-                }
-            } catch (e: Exception) {
-                if (e is TimeoutCancellationException) {
+            coroutineScope.async {
```

Comment:
you're still doing `runBlocking` so i think this still stalls EDT

Reply:
gotcha, my bad, i thought i had to use `runBlocking` to connect non coroutine with coroutine call, i removed `runBlocking`